src/app/features/tech-record/components/change-vehicle-type/change-vehicle-type.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Location } from '@angular/common';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { take } from 'rxjs';
+import { getOptionsFromEnum } from '@forms/utils/enum-map';
+import { MultiOptions } from '@forms/models/options.model';
+
+@Component({
+  selector: 'app-change-vehicle-type',
+  templateUrl: './change-vehicle-type.component.html',
+  styleUrls: ['./change-vehicle-type.component.scss']
+})
+export class ChangeVehicleTypeComponent implements OnInit, OnChanges {
+  constructor(
+    public dfs: DynamicFormService,
+    public globalErrorService: GlobalErrorService,
+    private technicalRecordService: TechnicalRecordService,
+    private location: Location
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(data => (this.vehicleTechRecord = data));
```

Comment:
unsure how much it matters but most other places we've done this sort of thing inside the oninit